File: c/leetcode/283. Move Zeroes.c


void moveZeroes(int *nums, int numsSize)
{
  int anchor = 0;
  int explorer = 0;

  while (explorer < numsSize)
  {
    if (nums[explorer] != 0)
    {
      if (explorer != anchor)
      {
        int temp = nums[anchor];
        nums[anchor] = nums[explorer];
        nums[explorer] = temp;
      }
      anchor++;
    }
    explorer++;
  }
}

File: python/leetcode/283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        anchor = 0
        explorer = 0
        
        while explorer < len(nums):
            
            if nums[explorer] != 0:
                if anchor != explorer:
                    temp = nums[explorer]
                    nums[explorer] = nums[anchor]
                    nums[anchor] = temp
                anchor += 1
            explorer += 1